solutions/gold/cf-510C.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: cf-510C
+source: CF
+title: Fox And Names
+author: Daniel Ge
+---
+
+## Solution
+The key idea here is to perform topological sort on the characters of the alphabet. 
+
+We find what must come before what by comparing consecutive strings. If string $name_i$ is lexicographically less than $name_{i+1}$, we compare only the first occurence of a different letter. In other words, we take the common prefix out and compare the first letters. For example, if we are given,
```

Comment:
> We find what must come before what by comparing consecutive strings.

This doesn't make sense

Reply:
> We find what must come before what by comparing consecutive strings. If string $name_i$ is lexicographically less than $name_{i+1}$, we compare only the first occurence of a different letter. In other words, we take the common prefix out and compare the first letters.

I think this can be worded better

Reply:
ok I rephrased everything to be more precise